plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/payment-methods/test/express-payment-methods.tsx

```diff
@@ -0,0 +1,137 @@
+	useEditorContext: mockEditorContext,
+} ) );
+
+// Button styles are disabled by default. We need to mock the express payment context
+// to enable them.
+jest.mock( '../express-payment/express-payment-context', () => {
+	return {
+		useExpressPaymentContext: jest.fn().mockReturnValue( {
+			showButtonStyles: true,
+			buttonHeight: '48',
+			buttonBorderRadius: '4',
+		} ),
+	};
+} );
+
+const mockExpressPaymentMethodNames = [ 'paypal', 'google pay', 'apple pay' ];
+
+const MockExpressButton = jest.fn( ( { name } ) => (
+	<div className="boo">{ `${ name } button` }</div>
+) );
+
+const MockEditorExpressButton = jest.fn( ( { name } ) => (
+	<div>{ `${ name } preview` }</div>
+) );
+
+const registerMockExpressPaymentMethods = () => {
+	mockExpressPaymentMethodNames.forEach( ( name ) => {
+		registerExpressPaymentMethod( {
+			name,
+			title: `${ name } payment method`,
+			description: `A test ${ name } payment method`,
+			gatewayId: 'test-express-payment-method',
+			paymentMethodId: name,
+			content: <MockExpressButton name={ name } />,
+			edit: <MockEditorExpressButton name={ name } />,
+			canMakePayment: () => true,
+			supports: {
+				features: [ 'products' ],
+			},
+		} );
+	} );
+	dispatch( PAYMENT_STORE_KEY ).__internalUpdateAvailablePaymentMethods();
+};
+
+const deregisterMockExpressPaymentMethods = () => {
+	mockExpressPaymentMethodNames.forEach( ( name ) => {
+		__experimentalDeRegisterExpressPaymentMethod( name );
+	} );
+};
+
+describe( 'Express payment methods', () => {
+	afterAll( () => {
+		jest.restoreAllMocks();
+	} );
+	describe( 'No payment methods available', () => {
+		it( 'should display no registered payment methods', () => {
+			render( <ExpressPaymentMethods /> );
+
+			const noPaymentMethods = screen.queryAllByText(
+				/No registered Payment Methods/
+			);
+			expect( noPaymentMethods.length ).toEqual( 1 );
+		} );
+	} );
+
+	describe( 'Payment methods available', () => {
+		beforeAll( () => {
+			registerMockExpressPaymentMethods();
+		} );
+		afterAll( () => {
+			deregisterMockExpressPaymentMethods();
+		} );
+		describe( 'In a frontend context', () => {
+			it( 'should display the element provided by paymentMethods.content', () => {
+				render( <ExpressPaymentMethods /> );
+				mockExpressPaymentMethodNames.forEach( ( name ) => {
+					const btn = screen.getByText( `${ name } button` );
+					expect( btn ).toBeVisible();
+				} );
+			} );
+			it( 'should pass the correct properties to the rendered element', () => {
+				render( <ExpressPaymentMethods /> );
+				mockExpressPaymentMethodNames.forEach( ( name ) => {
+					expect( MockExpressButton ).toHaveBeenCalledWith(
+						getExpectedExpressPaymentProps( name ),
+						{}
+					);
+				} );
+				// Expect some deprecation warnings
+				expect( console ).toHaveWarnedWith(
+					'isPristine is deprecated since version 9.6.0. Please use isIdle instead. See: https://github.com/woocommerce/woocommerce-blocks/pull/8110'
+				);
```

Comment:
Why? Is this part of the test?

Reply:
For some reason this gets triggered when `usePaymentMethodInterface` hook is called. I think it's because we use `get` [here](https://github.com/woocommerce/woocommerce/blob/feature/express-payment-improvements/plugins/woocommerce-blocks/assets/js/base/context/hooks/payment-methods/use-payment-method-interface.ts#L66). I've updated the comment to better reflect this.